src/bot.rs

```diff
@@ -0,0 +1,175 @@
+    from: telegram_bot_raw::types::chat::User
+) {
+    client.send_message(("Provo ad aggiungerti alla partita...", from.id).into());
+    // Check wether the user is already playing a game
+    if player_games.contains_key(&from.id) {
+        // The user can't join two games at the same time
+        client.send_message(("Non puoi unirti a più partite contemporaneamente", from.id).into());
+    } else {
+        if let Some(ch) = game_channel.get(&game_id) {
+            player_games.insert(from.id, game_id.clone());
+            ch.send(threading::ThreadMessage::AddPlayer(primitives::Player{id: from.id.into(), name: utils::get_user_name(&from.first_name, &from.last_name)})).unwrap();
+        } else {
+            client.send_message(("Gioco non trovato!", from.id).into());
+        }
+    }
+}
+
+fn init_game(
+    game_index: usize,
+    from: telegram_bot_raw::types::chat::User,
+    playable_games: &Vec<Box<dyn Game>>,
+    player_games: &mut HashMap<telegram_bot_raw::types::refs::UserId, String>,
+    game_channel: &mut HashMap<String, std::sync::mpsc::SyncSender<threading::ThreadMessage>>,
+    game_last_played: &mut HashMap<String, std::time::Instant>,
+    client: &mut Telegram
+) {
+    use threading::ThreadMessage;
+    let game_id = ulid::Ulid::new().to_string();
+    let (sender, reciever) = mpsc::sync_channel(10);
+    sender.send(ThreadMessage::AddPlayer(primitives::Player{id: from.id.into(), name: utils::get_user_name(&from.first_name, &from.last_name)})).unwrap();
+    player_games.insert(from.id, game_id.clone());
+    game_channel.insert(game_id.clone(), sender);
+    game_last_played.insert(game_id.clone(), std::time::Instant::now());
+    client.send_message((format!("Per invitare altre persone condividi questo link: https://t.me/giocoacartebot?start={}", game_id), from.id).into());
+    let game_tg_client = client.clone();
+    game_agent::new_agent(game_tg_client, game_index, playable_games, reciever);
+}
+
+fn handle_callback_query(
+    qry: telegram_bot_raw::types::callback_query::CallbackQuery,
+    playable_games: &Vec<Box<dyn Game>>,
+    player_games: &mut HashMap<telegram_bot_raw::types::refs::UserId, String>,
+    game_channel: &mut HashMap<String, std::sync::mpsc::SyncSender<threading::ThreadMessage>>,
+    game_last_played: &mut HashMap<String, std::time::Instant>,
+    client: &mut Telegram
+) {
+    use threading::*;
+    //let qry_id: String = qry.id.into;
+    //client.ack_callback_query(&format!("{}", qry.id));
+    let data: Vec<String> = qry.data.unwrap().split(":").map(|x| x.to_owned()).collect();
+    let command = data[0].as_str();
+    match command {
+        "init_game" => {
+            let params = data[1].clone();
+            let index = params.parse::<usize>().unwrap();
+            init_game(index, qry.from, playable_games, player_games, game_channel, game_last_played, client);
+        },
+        "start" => {
+            let player_id = qry.from.id;
```

Comment:
I would make a function for the entire case "start"

Reply:
Same as below